client/state/activity-log/utils.js

```diff
@@ -19,7 +19,8 @@ export const filterStateToApiQuery = ( filter ) => {
 		filter.group && { group: filter.group },
 		filter.notGroup && { not_group: filter.notGroup },
 		filter.name && { name: filter.name },
-		{ number: 1000 }
+		{ number: 1000 },
+		filter.queryPage && { page: filter.queryPage }
```

Comment:
We have to use `queryPage` because `filter.page` is used for paginating the UI, but at API level we use `page`